lib/entry/crossDCsStreamPersistence.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * Created by lilach.maliniak on 18/09/2016.
+ */
+var persistenceFormat = require('./../../common/PersistenceFormat');
+var qio = require('q-io/fs');
+//var sessionManager = require('./../SessionManager');
+var logger = require('../../common/logger').getLogger("CrossDCsStreamPersistence");
+var config = require('./../../common/Configuration');
+var ErrorUtils = require('./../utils/error-utils');
+var _ = require('underscore');
+var Q = require('q');
+var glob = require('glob');
+var path = require('path');
+var backendClient = require('../BackendClientFactory.js').getBackendClient();
+var PlaylistGenerator = require('./../playlistGenerator/PlaylistGenerator');
+const endListPendingEntriesFileName = "endListPending.json";
+const stoppedEntriesPath = config.get('stoppedEntriesPath');
+const kalturaTypes = require('./../kaltura-client-lib/KalturaTypes');
+
+// todo:
+// consult with Guy whether to keep playlist and just update on disk or read/write...
+// I think its better to keep cache of playlist and just write to disk each update.
+
+class CrossDCsStreamPersistence {
+
+    constructor() {
+
+        this._presentationEndTimeRegex = /("presentationEndTime":\s*)(\d+)/;
+        this._expireTimeRegex = /("expirationTime":\s*)(\d+)/;
+        this.stoppedEntriesFileName = path.join(stoppedEntriesPath, endListPendingEntriesFileName);
+        // note: default value of streamPersistenceTimer must be less than lowest play window size.
+        // assuming min play window size is 30 secs, the default persistenceCheckIntervalMillisec
+        // is 20 secs. In addition there is a time budget used as margin to make sure that even
+        // in very loaded system race condition chances will be very low.
+        this._persistenceCheckIntervalMillisec = config.get('persistenceCheckIntervalMillisec');
+        this._entriesToProcess = {};
+        this._playlistsData = {};
+        this._lastProcessedTimestampe = Date.now();
+        // _timeBudget purpose is to avoid race between packager and liveController, in case
+        // presentationEndTime value becomes past time before it is update by persistence loop.
+        this._timeBudget = 5000; // arbitrary time budget
+    }
+
+    streamPersistenceCheck(entryId, newEntry = false) {
+
+        let entryIsLive = false;
+        let ended = false;
+        let now = Date.now();
+        let nowStr = new Date(now).toLocaleString();
```

Comment:
why keep it as string?   dont' use presentationTimeStr and nowStr
also don't use LocaleString